tonic-build/src/server.rs

```diff
@@ -257,92 +257,54 @@ fn generate_trait_methods<T: Service>(
             quote!(&self)
         };
 
-        let method = match (
-            method.client_streaming(),
-            method.server_streaming(),
-            generate_default_stubs,
-        ) {
```

Comment:
I'd like to keep the current styles for the conditions about whether it is client streaming or server streaming or not. This refactoring is well designed to avoid reimplementing logic, but at the same time, I feel that it has become more difficult to grasp the outline of the processing from the perspective of code generation. Taking the trade-off into consideration, how about introducing `body_or_semicolon` to the conditional branch of `generate_default_stubs` to reduce the number of branches?

Reply:
Yeah I agree, refactored a bit more to look more like the older code. There's a bit of duplication here but I think it's a reasonable trade-off because it makes the branching easier to follow. What do you think?

Reply:
What I tried to say is that using `switch` with client streaming and/or server streaming, and then in the each branch using `body_or_semicolon` variable to handle the default stub option.

And, I would like to write `partial_sig` and `return_ty` inline directly without making them variables.

Reply:
Not sure I understand what you mean. I don't really see a point in doing something like `match (method.client_streaming(), method.server_streaming())` since those two settings don't impact one another.